### PR TITLE
Add MCS API

### PR DESCRIPTION
**Pull request by weeman1337**
---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: src/content-scanner/ContentScanner.ts [hunk 1]
```diff
@@ -0,0 +1,140 @@
+/*
```
Comment (estellecomment) [inline src/content-scanner/ContentScanner.ts:1]: not sure what the legal situation is, should we be having copyrighted New Vector code in tchap code ?

File: src/content-scanner/readScannerUrlConfig.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import SdkConfig from "matrix-react-sdk/src/SdkConfig";
+
+import { ContentScannerConfig } from "./ContentScannerConfig";
+
+/**
+ * Reads "content_scanner.url" from the config.
+ * Logs a warning if it is not present.
+ */
+export const readScannerUrlConfig = (): string => {
+    const contentScannerUrl = (SdkConfig.get() as unknown as ContentScannerConfig)?.content_scanner?.url;
```
Comment (jdauphant) [inline src/content-scanner/readScannerUrlConfig.ts]: On Tchap the content scanner is the base as the server. We have 17 differents homeservers different

File: src/content-scanner/readScannerUrlConfig.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
+/*
```
Comment (jdauphant) [inline src/content-scanner/readScannerUrlConfig.ts]: This file start by lower case but other file start by upper case

File: src/content-scanner/readScannerUrlConfig.ts (hunk 2 above)
Comment (jdauphant) [inline src/content-scanner/readScannerUrlConfig.ts]: The implementation should use the current homeserver of the user as content scanner endpoint and not an configurated value

File: src/content-scanner/ContentScanner.ts [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+/**
+ * Content scanner implementation that interacts with a Matrix Content Scanner.
+ * @see https://github.com/matrix-org/matrix-content-scanner
+ */
+export class ContentScanner implements ContentScannerInterface {
+    private static internalInstance: ContentScanner;
+
+    private mcsKey: PkEncryption = new global.Olm.PkEncryption();
+    private keySet = false;
```
Comment (estellecomment) [inline src/content-scanner/ContentScanner.ts]: "keySet" sounds like a set of keys. "hasKey" would be clearer to me.

File: src/content-scanner/ContentScanner.ts (hunk 1 above)
Comment (giomfo) [reply src/content-scanner/ContentScanner.ts:1]: @estellecomment we are using this header in all our contributions in Tchap mobile fork. It should be the same here. Let's talk in DM if need

File: src/content-scanner/readScannerUrlConfig.ts (hunk 3 above)
Comment (weeman1337) [reply src/content-scanner/readScannerUrlConfig.ts]: Yes. This file only exports a function with that name `readScannerUrlConfig` while the other files export components or classes like for instance `ContentScanner`.

This file is going to be removed anyway.

File: src/content-scanner/readScannerUrlConfig.ts (hunk 2 above)
Comment (weeman1337) [reply src/content-scanner/readScannerUrlConfig.ts]: Going to use the matrix server URL here.

File: src/content-scanner/ContentScanner.ts (hunk 4 above)
Comment (weeman1337) [reply src/content-scanner/ContentScanner.ts]: Done

**Comment by weeman1337** [comment]
Closed in favour of #132

**Pull request closed without merging**